src/components/Contacts/ContactDetails/ContactDetailsTab/EditContactModal/EditContactModal.tsx

```diff
@@ -0,0 +1,696 @@
+  fontWeight: 'bold',
+}));
+
+const ContactPrimaryPersonSelectLabel = styled(InputLabel)(() => ({
+  textTransform: 'uppercase',
+}));
+
+const ShowExtraText = styled(Typography)(() => ({
+  color: '#2196F3',
+  textTransform: 'uppercase',
+  fontWeight: 'bold',
+}));
+
+interface EditContactModalProps {
+  contact: ContactDetailsTabQuery['contact'];
+  isOpen: boolean;
+  handleOpenModal: (open: boolean) => void;
+}
+export const EditContactModal: React.FC<EditContactModalProps> = ({
+  contact,
+  isOpen,
+  handleOpenModal,
+}): ReactElement<EditContactModalProps> => {
+  const { t } = useTranslation();
+  console.log(contact);
+  const [personEditShowMore, setPersonEditShowMore] = useState<Array<string>>(
+    [],
+  );
+
+  const handleDateChange = (date: DateTime) => {
+    console.log(date.month);
+    console.log(date.day);
+    console.log(date.year);
+  };
+
+  const renderSocialsSection = (
+    person: ContactDetailsTabQuery['contact']['people']['nodes'][0],
+  ) => {
+    const socialAccounts = [
+      ...person.facebookAccounts.nodes.map((account) => ({
+        ...account,
+        type: 'facebook',
+        name: 'Facebook',
+      })),
+      ...person.twitterAccounts.nodes.map((account) => ({
+        ...account,
+        type: 'twitter',
+        name: 'Twitter',
+      })),
+      ...person.linkedinAccounts.nodes.map((account) => ({
+        ...account,
+        type: 'linkedin',
+        name: 'LinkedIn',
+      })),
+      ...person.websites.nodes.map((account) => ({
+        ...account,
+        type: 'website',
+        name: 'Website',
+      })),
+    ];
+    return socialAccounts.length > 0 ? (
+      <>
+        {socialAccounts.map((account, index) => (
+          <>
+            <ContactInputWrapper>
+              {index === 0 ? <ContactSocialIcon /> : null}
+              <Grid container spacing={3}>
+                <Grid item xs={6}>
+                  <TextField
+                    label={t('Username/URL')}
+                    value={account.value}
+                    fullWidth
+                  />
+                </Grid>
+                <Grid item xs={6}>
+                  <FormControl fullWidth>
+                    <InputLabel id="social-type-label">{t('Type')}</InputLabel>
+                    <Select
+                      labelId="socail-type-label"
+                      value={account.type}
+                      fullWidth
+                      readOnly
+                    >
+                      <MenuItem value={account.type}>{account.name}</MenuItem>
+                    </Select>
+                  </FormControl>
+                  <ContactEditDeleteIconButton>
+                    <DeleteIcon />
+                  </ContactEditDeleteIconButton>
+                </Grid>
+              </Grid>
+            </ContactInputWrapper>
+          </>
+        ))}
+      </>
+    ) : (
+      <>
+        <ContactInputWrapper>
+          <ContactSocialIcon />
+          <Grid container spacing={3}>
```

Comment:
This modal looks like a beast in the design, do you think any part of it will get used anywhere else? Or some of the sections could be their own components? Just thinking for working on this modal with all the stuff in it, it might help just mentally to split out some sections into components.